client/src/components/EditProfile/ProfilePhoto/UploadPhoto.tsx

```diff
@@ -1,20 +1,45 @@
-import React, { ChangeEvent, useState } from 'react';
+import React, { ChangeEvent, useEffect, useState } from 'react';
 import { Avatar, Box, Typography, Button } from '@material-ui/core';
 import DeleteIcon from '@material-ui/icons/Delete';
 
 import useStyles from './useStyles';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+import upload from '../../../helpers/APICalls/upload';
 
 const UploadPhoto: React.FC = (): JSX.Element => {
   const classes = useStyles();
+  const { updateSnackBarMessage } = useSnackBar();
 
-  const [selectedImage, setSelectedImage] = useState<File>();
+  const [selectedImage, setSelectedImage] = useState<File | undefined>();
 
-  const imageChange = (e: ChangeEvent<HTMLInputElement>) => {
-    if (e.target.files && e.target.files.length > 0) {
-      setSelectedImage(e.target.files[0]);
+  const handleImageChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const fileList = e.target.files;
+    if (!fileList) return;
+    setSelectedImage(fileList[0]);
+  };
+
+  const uploadImage = async () => {
+    if (selectedImage) {
+      const formData = new FormData();
+      formData.append('files', selectedImage);
+      formData.append('upload_preset', 'team-reindeer');
+
+      upload(formData).then((data) => {
+        if (data.error) {
+          updateSnackBarMessage(data.error.message);
+        } else {
+          console.error({ data });
+
+          updateSnackBarMessage('An unexpected error occurred. Please try again');
+        }
+      });
     }
   };
 
+  useEffect(() => {
+    uploadImage();
+  });
```

Comment:
Hi Lukman 👋🏼 This is the reason we upload the image multiple times. [`useEffect` docs](https://reactjs.org/docs/hooks-effect.html) mention: 
>Does useEffect run after every render? **Yes!** By default, it runs both after the first render and after every update. ([We will later talk about how to customize this](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects).)...

Hence, we need to make sure to only upload the image if a new image is selected in the UI. We can do this by running the effect *only if the `selectedImage` state chnages*. Check out the docs above to read more about how we can achieve this 👍🏼